File: Lab/SEM4/MPP/lab11/moto_c/src/Client/rest/LoggingInterceptor.cs
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Client.rest;

public static class LoggingInterceptor
{
    public static async Task<string> Intercept(
        HttpRequestMessage req,
        HttpResponseMessage res,
        string? body)
    {
        var outp = new StringBuilder();

        outp.AppendLine($"{req.Method} {req.RequestUri}");

        foreach (var h in req.Headers)
        {
            outp.AppendLine($"{h.Key}: {string.Join(", ", h.Value)}");
        }

        if (req.Content != null)
        {
            foreach (var h in req.Content.Headers)
            {
                outp.AppendLine($"{h.Key}: {string.Join(", ", h.Value)}");
            }
        }

        if (!string.IsNullOrWhiteSpace(body))
        {
            outp.AppendLine();
            outp.AppendLine("Request Body:");
            outp.AppendLine(body);
        }

        outp.AppendLine();
        outp.AppendLine("Sending request...");
        outp.AppendLine();

        outp.AppendLine("Response Status:");
        outp.AppendLine(((int)res.StatusCode).ToString());

        outp.AppendLine();

        outp.AppendLine("Response Body:");
        outp.AppendLine(await res.Content.ReadAsStringAsync());

        return outp.ToString();
    }
}

File: Lab/SEM4/MPP/lab11/moto_java/rest/src/main/java/org/zdroba/test/LoggingInterceptor.java
package org.zdroba.test;

import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class LoggingInterceptor {

    public static String intercept(HttpRequest req, HttpResponse<String> res, String body) {
        StringBuilder out = new StringBuilder();

        out.append(req.method())
                .append(" ")
                .append(req.uri())
                .append("\n");

        req.headers().map().forEach((k, v) ->
                out.append(k)
                        .append(": ")
                        .append(String.join(", ", v))
                        .append("\n")
        );

        if (body != null && !body.isBlank()) {
            out.append("\n")
                    .append("Request Body:\n")
                    .append(body)
                    .append("\n");
        }

        out.append("\n")
                .append("Sending request...\n\n")
                .append("Response Status:\n")
                .append(res.statusCode())
                .append("\n\n")
                .append("Response Body:\n")
                .append(res.body())
                .append("\n");

        return out.toString();
    }
}